src/beamlime/kafka/consumer.py

```diff
@@ -56,28 +57,26 @@ def make_consumer_from_config(
     *,
     topics: list[str],
     config: dict[str, Any],
-    instrument: str,
     group: str,
     unique_group_id: bool = True,
 ) -> Generator[kafka.Consumer, None, None]:
     """Create a Kafka consumer from a configuration dictionary."""
     if unique_group_id:
-        config['group.id'] = f'{instrument}_{group}_{uuid.uuid4()}'
-    with make_bare_consumer(
-        config=config,
-        topics=topic_for_instrument(topic=topics, instrument=instrument),
-    ) as consumer:
+        config['group.id'] = f'{group}_{uuid.uuid4()}'
+    with make_bare_consumer(config=config, topics=topics) as consumer:
         yield consumer
 
 
 @contextmanager
-def make_control_consumer(*, instrument: str) -> Generator[kafka.Consumer, None, None]:
+def make_control_consumer(
+    *, instrument: str
+) -> Generator[str, kafka.Consumer, None, None]:
     control_config = load_config(namespace='control_consumer', env='')
     kafka_downstream_config = load_config(namespace='kafka_downstream')
+    topic = stream_kind_to_topic(instrument=instrument, kind=StreamKind.BEAMLIME_CONFIG)
     with make_consumer_from_config(
-        topics=['beamlime_commands'],
+        topics=[topic],
         config={**control_config, **kafka_downstream_config},
-        instrument=instrument,
         group='beamlime_commands',
     ) as consumer:
-        yield consumer
+        yield topic, consumer
```

Comment:
This is a bit awkward, but I could not come up with a better way right now: `DataServiceBuilder` needs to know which topic is the config topic, since it needs a separate consumer to subscribe to. Otherwise it only has a `RouteByTopicAdapter`, which has a list of *all* topics.

I don't think this will cause much trouble because it will not be used in a ton of places.